sound/soc/sof/topology.c

```diff
@@ -870,12 +870,10 @@ static int sof_parse_uuid_tokens(struct snd_soc_component *scomp,
 	return found;
 }
 
-static int sof_parse_string_tokens(struct snd_soc_component *scomp,
-				   void *object,
+static int sof_parse_string_tokens(struct snd_soc_component *scomp, void *object,
 				   const struct sof_topology_token *tokens,
-				   int count,
-				   struct snd_soc_tplg_vendor_array *array,
-				   size_t offset)
+				   int count, struct snd_soc_tplg_vendor_array *array, int offset,
+				   struct snd_sof_tuple **tuples, int *num_tuples)
 {
```

Comment:
7 params, can we make it less ?

Reply:
No, we want to keep the existing function so as to minimize the amount of changes. I don't think it looks so bad to be honest

Reply:
as you know it is not a good practice to set many params in function call. Nice to have.

Reply:
if params count is less than 5 on X86, they can be stored in registers for function call else they will be stored in call stack, so some technical books recommend less params.